### ISSUE TITLE
Table E (Outputs) html summary doesn't match Table E Excel output

### ISSUE THREAD

**Issue by aturling** [EPSCoR/ERCore#18]
The Table E html page (reporting/outputs) has two headers: "Cumulative" and "Reporting Period". Below the "Cumulative" header, it displays patents, proposals, etc. but at the bottom of the page there's a "Reporting Period" header with no content below it. It's showing this way for our site and the dev-ercore site. I can't tell if this is a bug that's somehow just affecting our local install though because it could be that the dates entered in all of the dev-ercore content don't match up with the reporting period. Is this page working correctly on other production sites?

The weird thing is that on our site, even though the counts are displayed under the Cumulative header, they are actually the Reporting Period counts (they agree with the Reporting Period column counts on the Table E Excel output).

**Comment by iserna** [comment]
Yes, this is something we have noticed in NM on our live and dev sites.  I imagine it is the same for all other jurisdictions.

**Comment by cjallen-epscor** [comment]
Table E labels are incorrect.

You are right, the 'Cumulative' and 'Reporting Period' need to be switched.
From what I understand, 'Cumulative' is always called in the database; all content. 

The admin views are more concerned, for us at least, about 'Reporting Period'.

I can make the switch and have it in the next 3.2-dev branch.

**Comment by aturling** [comment]
Actually is there an easy way to check if the content dates are within the date range of the lifetime of the EPSCoR grant? We have a few people who entered in accomplishments that occurred before the grant award date, so they're being counted in Table E (in the cumulative column) when they shouldn't be included.

**Comment by iserna** [comment]
From http://dev-ercore.nmepscor.net./proposals you can filter by date.  

How to find this view: 
From the top/black navigation go to EPSCoR > EPSCoR Content > Proposals or Grants.

We do have an award that falls outside of the grant timeframe and I am not sure right now if it is counted in the Cumulative or not.  I will confirm and let you know.

**Comment by aturling** [comment]
On our site, "cumulative" = everything entered into the site that doesn't fall under the current reporting period.

I think I figured out what's going on with the Table E html output: in the html() function, the nested for loops store stuff to $output using the current category and/or row as array keys. But both "cumulative" and "reporting period" contain the same categories and rows, so on the second iteration of the outermost for loop, the $output values are being overwritten. That's why the page only contains the reporting period data, as that's populated in the second iteration of the loop (overwriting the cumulative data). To fix this, the current $range_label (cumulative vs. reporting period) can be added to the key. I changed these two lines and the html page matches the Excel output now:

in pages/outputs.inc:

Line 275:
Old: `$output[$cat_key . '-label'] = $this->h2($cat_key);`
New: `$output[$range_label . '-' . $cat_key . '-label'] = $this->h2($cat_key);`

Line 294:
Old: `$output["$cat_key-$row_key-list"] = $list;`
New: `$output["$range_label-$cat_key-$row_key-list"] = $list;`

**Comment by iserna** [comment]
Amy, I just added/updated my local install.  I will test it out and let you know if I find anything odd or if I have any questions.

Thank you!

**Comment by cjallen-epscor** [comment]
@aturling - awesome! I placed this patch on our local NM install and it seems to work great!

I made some minor styling modifications which can be seen here:
http://dev-ercore.nmepscor.net/reporting/outputs

I placed the Reporting Period date on the 'Reporting Date' area and added a rule to break up the areas between Cumulative and Reporting Period.

@aturling and my slight mods can be viewed at ~~https://github.com/EPSCoR/ERCore-3.1/commit/ca60db4c2c092eacac1e1a5a4030574d6fa3122e~~  https://github.com/EPSCoR/ERCore-3.1/commit/a147db10ef8abc831daeff8a7170e2007af1bb59
